Packages: Import photon package

sirreal opened the pull request:
#### Changes proposed in this Pull Request

* Pulls in photon source
* ~Preps a 3.0.0 version~
* Update to match Calypso style (es6, Jest…)

#### Testing instructions

* Tests pass?
* Shrinkwrap changes look correct?
* Photon works correctly in Calypso (reader, theme showcase…)

TooTallNate committed:
initial commit

TooTallNate committed:
package: set "author" to Automattic, add self as "contributor"

TooTallNate committed:
test: initial tests

TooTallNate committed:
index: prevent Photon URL "inception"

TooTallNate committed:
add LICENSE file

TooTallNate committed:
add Readme.md file

TooTallNate committed:
add Makefile, with browserify make target for browsers

TooTallNate committed:
dist: precompile

TooTallNate committed:
index: make `secure: false` work

TODO: add tests for that

TooTallNate committed:
remove test.js file

TooTallNate committed:
idnex: statically hash the subdomain based on the URL

As recommended in the Photon documentation.

TooTallNate committed:
test: fix test case after server distribution logic

TooTallNate committed:
dist: recompile

retrofox committed:
ignoring node_modules folders

TooTallNate committed:
Readme: add npm install instructions

TooTallNate committed:
test: add another test

TooTallNate committed:
v0.0.1

TooTallNate committed:
v1.0.0

TooTallNate committed:
Readme: use Cloudup image in examples

It's better than an all white image on an all white background
like the image was before

TooTallNate committed:
package: update "debug" to v1.0.0

TooTallNate committed:
package: be loose with the `debug` version

I'm the maintainer now so it's not going to break.

TooTallNate committed:
v1.0.1

TooTallNate committed:
index: remove "port 80" debug note

It's actually inaccurate. Photon checks HTTPS port 443 as
well as standard HTTP port 80 access.

TooTallNate committed:
Readme: add "Terms of Service" section

Matches the text in the official Photon docs.
It's a good idea to emphasize the terms of use so that users read it.

TooTallNate committed:
v1.0.2

TooTallNate committed:
dist: recompile

TooTallNate committed:
package: update "browserify" to v4.2.0

TooTallNate committed:
package: update "browserify" to v6.1.0

TooTallNate committed:
dist: recompile

TooTallNate committed:
Merge pull request #2 from Automattic/fix/photon-ception

fixes photonception issue

TooTallNate committed:
dist: recompile

TooTallNate committed:
v1.0.3

TooTallNate committed:
Merge pull request #3 from Automattic/fix/querystring-args-and-inception

Fix how querystrings are dealt with for photon hosts, protcolless urls.

TooTallNate committed:
package: update "browserify" to v8.1.1

TooTallNate committed:
package: allow any "debug" v2

TooTallNate committed:
Release 1.0.4

blowery committed:
Add support for the ssl parameter to fetch images from https URLs

Closes #5.

blowery committed:
Return `null` for external URLs that contain querystrings

Photon doesn't support them.

Closes #4.

Squashed commit of the following:

commit 7a5e4049d05e865b9daa280df80369b8219a0c77
Author: Ben Lowery <ben.lowery@automattic.com>
Date:   Mon Feb 1 10:42:36 2016 -0500

    Return null for external URLs that contain querystrings

commit b7fd525f82d19a8208d2ff7e293c7fccebcd7b12
Author: Ben Lowery <ben.lowery@automattic.com>
Date:   Wed Apr 15 14:28:59 2015 -0400

    Don't attempt to photon URLs that use a querystring.

    Photon doesn't support them.

TooTallNate committed:
package: update "mocha" to v2.4.5

TooTallNate committed:
add .travis.yml file for Travis-CI

TooTallNate committed:
Readme: add Travis-CI badge

TooTallNate committed:
Release 2.0.0

TooTallNate committed:
dist: recompile

DanReyLop committed:
Fix handling of "blob:" URLs on NodeJS 7+ (#6)

* Fix handling of "blob:" URLs on NodeJS 7+

This PR adds a test for transforming a `blob:` URL, that failed in
NodeJS 7 and 8 due to a change in the built-in `url.format` logic.
It also fixes the issue and adds targets for NodeJS 7, 8 and 9
for TravisCI.

sirreal committed:
Prep 2.0.1 for release (#7)

* Update dependencies
* Build dist/photon.js
* Add history entry
* v2.0.1 bump

sirreal committed:
Update travis Node versions (#8)

* Update node/npm test versions

* Clean up Travis config

* Update package-lock for node@10 and npm@6.1

sirreal committed:
Prepare imported module for Calypso repo

sirreal commented on packages/photon/Makefile:
```diff
@@ -0,0 +1,36 @@
+
```
👍

sirreal commented on packages/photon/History.md:
```diff
@@ -0,0 +1,65 @@
+3.0.0 / 2018-12-05
```
Update before release…

sirreal commented on packages/photon/.eslintrc.js:
```diff
@@ -0,0 +1,8 @@
+/** @format */
+
+module.exports = {
+	extends: '../../.eslintrc.js',
```
Not needed.

sirreal commented on packages/photon/LICENSE:
```diff
@@ -0,0 +1,24 @@
+(The MIT License)
```
MIT? Does this need to be GPL or GPLv2?

Remove the license and use the package.json property

sirreal commented:
Rebased with a few changes I'd noted and a fresh `update-deps`. Ready for review.

sirreal commented:
Rebased with a few changes I'd noted and a fresh `update-deps`. Ready for review.

It looks like the photon may need some work for #29559, I'd like to get this landed.

sirreal commented:
Desktop tests are failing, seem to be exposing a problem with the package or with the desktop setup. We'll need to investigate…

blowery commented:
@adlk any ideas on the failing desktop build here?
@sirreal I think we fixed some issues with the desktop build after this PR was created. Maybe we just need to drop the branch with the same name from the desktop repo and let it get re-created before the test run?

blowery commented:
(just did the branch dance with desktop. we'll see)

blowery commented:
new branch rebuild didn't help :/

sirreal commented:
I've rebased and removed the desktop test branch, let's see if it's happy now.

sirreal commented:
This seems to be working just fine in the themes showcase where it's used:

https://hash-36207191d5e75c4537ae2e55c118912e2e879160.calypso.live/themes

However e2e tests are failing, seemingly unrelated.

bsessions85 commented:
e2e tests failures are unrelated. We've been having some issues I am working on fixing right now.  This should be good

sirreal commented:
Needs to be updated after photon 3 release with https://github.com/Automattic/photon.js/pull/10

blowery commented on .eslintignore:
```diff
@@ -3,7 +3,7 @@
 /build/
 /docs/
 /public/
-/packages/*/dist/**/*
```
can drop this change i think

sirreal replied:
Dropped the change. Should be synonymous: ignore the dir, or ignore everything in all the dirs in the dir 🙂

blowery commented on packages/photon/package-lock.json:
```diff
@@ -0,0 +1,50 @@
+{
```
can drop the package lock entirely and add a .npmrc that disallows package locks

sirreal replied:
28b76fb770b98ac19fc500a54acdc78cd136835c

blowery commented on packages/photon/package.json:
```diff
@@ -0,0 +1,37 @@
+	"version": "2.0.1",
+	"description": "JavaScript library for the WordPress.com Photon image manipulation service",
+	"main": "dist/cjs/index.js",
+	"module": "dist/esm/index.js",
```
should add `sideEffects: false`

sirreal replied:
4f03cbd4743db2f8f83a6745d6d9adff634df87e

sirreal committed:
Remove photon makefile

sirreal committed:
Remove package lock

sirreal committed:
Add sideEffects: false

sirreal commented on packages/photon/Makefile:
```diff
@@ -0,0 +1,36 @@
+
```
bfad93e4bb212d8021b7eeaea091abed624db591

sirreal committed:
Revert eslintignore change

sirreal committed:
Drop @format

sirreal commented:
Oh hey, the desktop tests passed! Let's move ahead with this 🤞

blowery commented:
@sirreal what do you think, do we want to update calypso to use it too? Or just import it for this PR?

sirreal commented:
Let's use it, I'll update this branch and give it a test 👍

sirreal committed:
Use photon package

sirreal committed:
Update deps

sirreal commented:
One note: 

With

- https://github.com/Automattic/photon.js/pull/10
- https://github.com/Automattic/photon.js/pull/11

Photon repo is ahead of this repo. However, that published version was deprecated, the latest tag was rolled back, and the enhancement may be stalled indefinitely:

- https://github.com/Automattic/photon.js/pull/12
- pMz3w-9qA-p2

I'm not sure whether it's worth pulling those changes into this branch or what. That will be a concern for the next release because it may need to be v4. It would be nice to get conclusion to pMz3w-9qA-p2.

sirreal commented:
- Added `file:` ref to use the package.
- Updated deps `npm run update-deps`
- Tested reader, themes, media editor, photon seems to be working well 👍

sirreal committed:
Remove format pragma

blowery reviewed (approved):
This looks good!

blowery commented:
Looks like the package.json just switched from tabs to spaces (or vice versa)

blowery committed:
switch package.json back to spaces

ockham commented:
Seems like this broke jurassic.ninja's Jetpack blocks build step, since that didn't run `npm ci && npm run build-packages` but was still using `lerna bootstrap --ci`.

Found by @jeherve and fixed by @oskosk. Convo: p1550870061189900-slack-jetpack-gutenberg
